storage/innobase/srv/srv0start.cc

```diff
@@ -2786,6 +2889,14 @@ dberr_t srv_start(bool create_new_db, const std::string &scan_directories) {
       return (srv_init_abort(err));
     }
 
+    if (srv_undo_log_encrypt) {
+      err = check_mk_and_keyring_encrypt_exclusion_for_undo(false, nullptr);
```

Comment:
I think it makes sense to directly call this function and read the value of srv_undo_log_encrypt inside it